qgsquick/from_qgis/plugin/qgsquickfeatureform.qml

```diff
@@ -391,36 +392,57 @@ Item {
       anchors {
         left: parent.left
         right: parent.right
-        leftMargin: 12 * QgsQuick.Utils.dp
+        leftMargin: form.style.fields.outerMargin
+        rightMargin: form.style.fields.outerMargin
       }
 
-      Label {
-        id: fieldLabel
-
-        text: Name ? qsTr(Name) : ''
-        font.bold: true
-        color: ConstraintSoftValid && ConstraintHardValid ? form.style.constraint.validColor : form.style.constraint.invalidColor
-      }
-
-      Label {
-        id: constraintDescriptionLabel
+      Item {
+        id: labelPlaceholder
+        height: fieldLabel.height + constraintDescriptionLabel.height + 2 * form.style.fields.sideMargin
         anchors {
           left: parent.left
           right: parent.right
-          top: fieldLabel.bottom
+          topMargin: form.style.fields.sideMargin
+          bottomMargin: form.style.fields.sideMargin
+        }
+
+        Label {
+          id: fieldLabel
+
+          text: Name ? qsTr(Name) : ''
+          color: ConstraintSoftValid && ConstraintHardValid ? form.style.constraint.validColor : form.style.constraint.invalidColor
+          leftPadding: form.style.fields.sideMargin
+          font.pointSize: form.style.fields.labelPointSize
+          horizontalAlignment: Text.AlignLeft
+          verticalAlignment: Text.AlignVCenter
+          anchors.topMargin: form.style.fields.sideMargin
+          anchors.top: parent.top
+        }
+
+        Label {
+          id: constraintDescriptionLabel
+          anchors {
+            left: parent.left
+            right: parent.right
+            top: fieldLabel.bottom
+            leftMargin: form.style.fields.sideMargin
+          }
+
+          text: ConstraintDescription ? qsTr(ConstraintDescription) : ''
+          visible: !ConstraintHardValid || !ConstraintSoftValid
+          height: visible ? undefined : 0
```

Comment:
:flushed: I know this is not really a part of the PR, but how does this work at all? :-) Do you have a screenshot of how things look when a constraint is not valid?

Reply:
Made one more commit to get rid of an empty space if a constrain description is empty
Also committed forgotten tab title font size to be able to set it from styling

<img width="752" alt="Screenshot 2021-02-22 at 16 46 02" src="https://user-images.githubusercontent.com/6735606/108732962-56321400-752e-11eb-9a16-e0bb33224862.png">